internal/install/segment/client.go

```diff
@@ -0,0 +1,121 @@
+package segment
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+const (
+	eventName = "VirtuosoCLIInstall"
+)
+
+type EventType string
+
+var EventTypes = struct {
+	InstallStarted          EventType
+	AccountIDMissing        EventType
+	APIKeyMissing           EventType
+	RegionMissing           EventType
+	UnableToConnect         EventType
+	UnableToFetchLicenseKey EventType
+	AbleToFetchLicenseKey   EventType
+	UnableToOverrideClient  EventType
+}{
+	InstallStarted:          "InstallStarted",
+	AccountIDMissing:        "AccountIDMissing",
+	APIKeyMissing:           "APIKeyMissing",
+	RegionMissing:           "RegionMissing",
+	UnableToConnect:         "UnableToConnect",
+	UnableToFetchLicenseKey: "UnableToFetchLicenseKey",
+	AbleToFetchLicenseKey:   "AbleToFetchLicenseKey",
+	UnableToOverrideClient:  "UnableToOverrideClient",
+}
+
+var (
+	embedded embed.FS
+)
+
+type Segment struct {
+	analytics.Client
+}
+
+func New() *Segment {
+	writeKey, err := getWriteKey()
```

Comment:
We should handle the case when we have no write key defined, and I don't think this should even be consider a warning. I would just make that a debug statement.